templates/interfaces.j2

```diff
@@ -1,49 +1,127 @@
 
-{% for interface in bridges %}
+{% if network_management_clear_bridges %}
+# Delete all network bridges
+pre-up for br in `ovs-vsctl list-br`; do ovs-vsctl --if-exists "$br"; done
+{% endif %}
+
+################################################################################
+# Configure all network interfaces                                             #
+################################################################################
+{% for interface in interfaces %}
+{% if interface.bring_up|default(network_management_default_bring_up) %}
+auto {{ interface.name }}
+{% endif -%}
+
+iface {{ interface.name }} inet {{ interface.type|default(network_management_default_type) }}
+{% if interface.type|default(network_management_default_type) == "dhcp" %}
+	hostname {{ interface.hostname|default(ansible_hostname) }}
+	{% if interface.leasetime is defined %}leasetime {{ interface.leasetime }}{% endif %}
+	{% if interface.metric is defined %}metric {{ interface.metric }}{% endif %}
+{% elif interface.type|default(network_management_default_type) == "static" -%}
+
+{% for ip in interface.ips %}
+	{% if not loop.first %}
+
+iface {{ interface.name }} inet {{ interface.type|default(network_management_default_type) }}
+	{% endif -%}
+	address {{ ip | ipaddr('address') }}
+	network {{ ip | ipaddr('network') }}
+	netmask {{ ip | ipaddr('netmask') }}
+
+{% endfor %}
+{% endif %}
+{% endfor %}
+
+################################################################################
+# Configure all network bridges                                                #
+################################################################################
+# Bring up all ports for all bridges
+{% for bridge in bridges %}
 {% for port in interface.ports | default({}) %}
 auto {{ port.port }}
 iface {{ port.port }} inet manual
 
 {% endfor %}
 {% endfor %}
 
+# Bring all openvswitch bridges up and configure ovs
 {% for interface in bridges %}
 auto {{ interface.name }}
-{% if interface.dhcp | default(False) %}
+{% if interface.type | default(network_management_default_type) == "dhcp"%}
 iface {{ interface.name }} inet dhcp
-{% elif interface.static is defined %}
+	hostname {{ interface.hostname|default(ansible_hostname) }}
+	{% if interface.leasetime is defined %}leasetime {{ interface.leasetime }}{% endif %}
+	{% if interface.metric is defined %}metric {{ interface.metric }}{% endif %}
+{% elif interface.type | default(network_management_default_type) == "static" %}
 iface {{ interface.name }} inet static
-	address {{ interface.static | ipaddr('address') }}
-	network {{ interface.static | ipaddr('network') }}
-	netmask {{ interface.static | ipaddr('netmask') }}
-	broadcast {{ interface.static | ipaddr('broadcast') }}
+	address {{ interface.static[0] | ipaddr('address') }}
+	network {{ interface.static[0] | ipaddr('network') }}
+	netmask {{ interface.static[0] | ipaddr('netmask') }}
+{% if interface.broadcast is defined %}
+	broadcast {{ interface.broadcast }}
```

Comment:
Can't this be calculated from the values above?

Reply:
debian uses ip defaults here, but you can define just a smaller subnet as broadcast address.